site/content/docs/howto/supervisor/configure-supervisor-federationdomain-idps.md

```diff
@@ -0,0 +1,396 @@
+string constants and string list constants for your transformation pipeline.
+
+For example, if there is a special username or group name which will be used for comparisons in your expressions,
+then you might like to declare it as a string constant. If there is a special list of usernames or group names
+which will be used for comparisons then you might like to declare the list as a constant.
+
+Constants are available in every expression of the pipeline.
+
+### Transformation pipelines `examples`
+
+Because the pipelines of expressions may behave differently based on their inputs, you may also optionally configure
+`examples` to demonstrate how a pipeline is expected to behave for various possible input scenarios. These examples
+act as living documentation for your fellow administrators, and also act as unit tests for your CEL expression code.
+
+Each example declares inputs for the whole pipeline of expressions, and also declares the expected results of the
+entire pipeline running on those inputs. The inputs are examples of the username and list of group names that might
+be determined by the related OIDCIdentityProvider, ActiveDirectoryIdentityProvider, or LDAPIdentityProvider resource.
+The expected outputs are the username and list of group names, or the authentication rejection, for which your pipeline
+should result upon the given inputs.
+
+If any example does not behave as expected, Pinniped will mark the whole FederationDomain with an error in
+its `status` and users will not be allowed to use the FederationDomain to authenticate until the error is corrected.
+
+### Putting it all together: an example of a transformation pipeline configuration
+
+The following example is contrived to demonstrate every feature of the `transformations` configuration.
+It is likely more complex than a real configuration.
+
+```yaml
+kind: FederationDomain
+apiVersion: config.supervisor.pinniped.dev/v1alpha1
+metadata:
+  name: demo-federation-domain
+  namespace: pinniped-supervisor
+spec:
+  issuer: https://issuer.example.com/demo-issuer
+  tls:
+    secretName: my-federation-domain-tls
+  identityProviders:
+  - displayName: ActiveDirectory for Admins
+    objectRef:
+      apiGroup: idp.supervisor.pinniped.dev
+      kind: ActiveDirectoryIdentityProvider
+      name: ad-for-admins
+    transforms:
+       constants:
+         - name: prefix
+           type: string
+           stringValue: "ad:"
+         - name: onlyIncludeGroupsWithThisPrefix
+           type: string
+           stringValue: "kube/"
+         - name: mustBelongToOneOfThese
+           type: stringList
+           stringListValue: [ "kube/admins", "kube/developers", "kube/auditors" ]
+         - name: additionalAdmins
+           type: stringList
+           stringListValue: [ "ryan@example.com", "ben@example.com", "josh@example.com" ]
+       expressions:
+         # This expression runs first, so it operates on unmodified usernames and groups
```

Comment:
Noting quick that we saw the comments wrap so we should shorten the lines:

<img width="805" alt="Screenshot 2023-09-12 at 1 33 18 PM" src="https://github.com/vmware-tanzu/pinniped/assets/280512/ac6fdb19-8dd1-4bb0-a852-db523482d291">

Reply:
Updated